onchain/src/quests/template_quest.cairo

```diff
@@ -0,0 +1,69 @@
+use art_peace::templates::interfaces::{
+    ITemplateStoreDispatcher, ITemplateStoreDispatcherTrait, ITemplateVerifierDispatcher,
+    ITemplateVerifierDispatcherTrait, TemplateMetadata
+};
+
+#[starknet::contract]
+pub mod TemplateQuest {
+    use starknet::{ContractAddress, get_caller_address};
+    use art_peace::{IArtPeaceDispatcher, IArtPeaceDispatcherTrait};
+    use art_peace::quests::{IQuest, QuestClaimed};
+
+    #[storage]
+    struct Storage {
+        art_peace: IArtPeaceDispatcher,
+        reward: u32,
+        claimed: LegacyMap<ContractAddress, bool>,
+    }
```

Comment:
You'll want to add a constructor to setup the reward, contract addresses, and dispatchers as well.

Reply:
this make sense, i will some thing like this   

```
  #[derive(Drop, Serde)]
    pub struct TemplateQuestInitParams {
        pub art_peace: ContractAddress,
        pub reward: u32,
        pub template_store: ContractAddress
    }
```
then initiated them inside the constructor